api/src/org/apache/cloudstack/api/command/admin/account/CreateAccountCmd.java

```diff
@@ -196,5 +204,8 @@ private void validateParams() {
         if(StringUtils.isEmpty(getPassword())) {
             throw new ServerApiException(ApiErrorCode.PARAM_ERROR, "Empty passwords are not allowed");
         }
+        if (getAccountType() == null && getRoleId() == null) {
```

Comment:
Consider splitting these checks to give a more precise error message to the end user.  Also, shouldn't check that ``roleId`` is greater than ``0L``?